clang/include/indexstore/indexstore.h

```diff
@@ -133,6 +133,11 @@ typedef void *indexstore_t;
 INDEXSTORE_PUBLIC indexstore_t
 indexstore_store_create(const char *store_path, indexstore_error_t *error);
 
+INDEXSTORE_PUBLIC indexstore_t
+indexstore_store_create_with_prefix_mapping(
+    const char *store_path, const char **PrefixMappings,
+    size_t NumMappings,indexstore_error_t *error);
```

Comment:
To be more flexible about evolving the C API I'd recommend to avoid adding an ad-hoc `indexstore_store_create_with_prefix_mapping` (ideally we'd avoid the situation of adding new variants of that function whenever we find a need to pass something additional).

I'd suggest to introduce a new opaque object, like `indexstore_creation_options_t`, with C functions to set values for it, then introduce a `indexstore_store_create_with_options` function.

Reply:
You should also increment the `INDEXSTORE_VERSION_MINOR` value in the header.

Reply:
SGTM, I'll go ahead and do that, that makes sense for future proofing.